ts/src/pro/binance.ts

```diff
@@ -989,59 +978,105 @@ export default class binance extends binanceRest {
-        [ type, params ] = this.handleMarketTypeAndParams ('watchTickers', market, params);
-        let subType = undefined;
-        [ subType, params ] = this.handleSubTypeAndParams ('watchTickers', market, params);
-        if (this.isLinear (type, subType)) {
-            type = 'future';
-        } else if (this.isInverse (type, subType)) {
-            type = 'delivery';
+        const newTickers = await this.watchMultiTickerHelper ('watchTickers', channelName, symbols, params);
+        if (this.newUpdates) {
+            return newTickers;
         }
-        const options = this.safeValue (this.options, 'watchTickers', {});
-        let name = this.safeString (options, 'name', 'ticker');
-        name = this.safeString (params, 'name', name);
-        params = this.omit (params, 'name');
-        let wsParams = [];
-        let messageHash = 'tickers';
-        if (symbols !== undefined) {
-            messageHash = 'tickers::' + symbols.join (',');
+        return this.filterByArray (this.tickers, 'symbol', symbols);
+    }
+
+    async watchBidsAsks (symbols: Strings = undefined, params = {}): Promise<Tickers> {
+        /**
+         * @method
+         * @name gate#watchBidsAsks
+         * @see https://www.gate.io/docs/developers/apiv4/ws/en/#best-bid-or-ask-price
+         * @see https://www.gate.io/docs/developers/apiv4/ws/en/#order-book-channel
+         * @description watches best bid & ask for symbols
+         * @param {string[]} symbols unified symbol of the market to fetch the ticker for
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/#/?id=ticker-structure}
+         */
+        const result = await this.watchMultiTickerHelper ('watchBidsAsks', 'bookTicker', symbols, params);
+        if (this.newUpdates) {
+            return result;
         }
-        if (name === 'bookTicker') {
-            if (marketIds === undefined) {
-                throw new ArgumentsRequired (this.id + ' watchTickers() requires symbols for bookTicker');
-            }
-            // simulate watchTickers with subscribe multiple individual bookTicker topic
-            for (let i = 0; i < marketIds.length; i++) {
-                wsParams.push (marketIds[i].toLowerCase () + '@bookTicker');
+        return this.filterByArray (this.tickers, 'symbol', symbols);
+    }
+
+    async watchMultiTickerHelper (methodName, channelName: string, symbols: Strings = undefined, params = {}) {
+        await this.loadMarkets ();
+        symbols = this.marketSymbols (symbols, undefined, true, false, true);
+        let firstMarket = undefined;
+        let marketType = undefined;
+        const symbolsDefined = (symbols !== undefined);
+        if (symbolsDefined) {
+            firstMarket = this.market (symbols[0]);
+        }
+        [ marketType, params ] = this.handleMarketTypeAndParams (methodName, firstMarket, params, undefined);
```

Comment:
why provide undefined as the default value? this symbol/type inference logic is confusing

Reply:
there was the following reason to me:
if you call `watchTickers (undefined, {subType: 'linear'})` it should fetch all tickers from binance-usdm (swap and future symbols), right? but it didn't work, it ended up with `marketType = 'spot' & subType=linear`, and having `marketType` defined to `spot` caused misbehavior down the rest of the part, where `marketType` is being checked. so, when someone runs `fetchTickers (undefined, {subType: 'linear'})` imo it expresses what user wants (to get all tickers from `binance-usdm`). it's not ok to call `fetchTickers (undefined, {defaultType:'swap', subType: 'linear'}` because then you will get only linear `swap` symbols and not linear futures, and then you will need separate call with {defaultType:'future'} (and each fetchTickers call takes 40 weight), so it was wasted call in vain, while you can get swap+future linear all symbols in one call with `subType: linear` (because binance have separation by linear / inverse, and not swap / future)

Reply:
isn't this enough?

```Typescript
        [ marketType, params ] = this.handleMarketTypeAndParams (methodName, firstMarket, params);
        let subType = undefined;
        [ subType, params ] = this.handleSubTypeAndParams (methodName, firstMarket, params);
        let rawMarketType = marketType;
        if (this.isLinear (marketType, subType)) {
            rawMarketType = 'future';
        } else if (this.isInverse (marketType, subType)) {
            rawMarketType = 'delivery';
        }
```

Reply:
you can't even call `{defaulType: undefined, subType:'linear'}` because it would still default to `spot`. so, that's why i've tried to avoid "defaulting" to spot **when subType param was detected**.

Reply:
ah, wait, sorry, i was disoriented when takling on `fetchTickers`. forget REST calls, i wanted to mean `watchTickers`

Reply:
the other things what i mentioned stays same, so for example, you want to run `watchTickers (undefined, {subType: 'linear'})` it should be able to work, right?

Reply:
type has priority over the subType in this case, if type is provided and it's spot ignore subtype, if type = swap just assume subType as linear unless subType is provided